backup/backup.go

```diff
@@ -311,18 +311,37 @@ func (b *Backup) writeBackupRemote() {
 		log.Fatalf("[ERR] Could not read compressed file!: %v", err)
 	}
 
-	// Create the params to pass into the actual uploader
-	params := &s3manager.UploadInput{
-		Bucket: &b.Config.S3Bucket,
-		Key:    &b.RemoteFilePath,
-		Body:   bytes.NewReader(localFileContents),
-	}
+	if len(b.Config.S3Bucket) > 1 {
+		s3Conn := session.New(&aws.Config{Region: aws.String(string(b.Config.S3Region))})
```

Comment:
Can this be made into a dedicated function like `writeBackupRemoteS3()` and below a similar function called `writeBackupRemoteGoogleStorage()`?  Then the logic becomes very easy to follow (and testable!).